```
Parsing Kconfig tree in /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/Kconfig
Traceback (most recent call last):
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfig.py", line 201, in <module>
    main()
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfig.py", line 32, in main
    kconf = Kconfig(args.kconfig_root, warn_to_stderr=False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 1011, in __init__
    check_dep_loop_sym(sym, False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6123, in _check_dep_loop_sym
    loop = _check_dep_loop_choice(sym.choice, sym)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6156, in _check_dep_loop_choice
    loop = _check_dep_loop_sym(sym, True)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6116, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6116, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6116, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6116, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6116, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6116, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6120, in _check_dep_loop_sym
    return _found_dep_loop(loop, sym)
  File "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/kconfig/kconfiglib.py", line 6222, in _found_dep_loop
    raise KconfigError(msg)
kconfiglib.KconfigError: 
Dependency loop
===============

SPI_6 (defined at drivers/spi/Kconfig:165), with definition...

config SPI_6
	bool
	prompt "SPI port 6" if SPI_6 && SPI
	depends on SPI_6 && SPI
	help
	  Enable SPI controller port 6.

...depends again on SPI_6 (defined at drivers/spi/Kconfig:165)
CMake Error at /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/cmake/kconfig.cmake:161 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/cmake/app/boilerplate.cmake:395 (include)
  CMakeLists.txt:10 (include)

```

--- a/boards/arm/nucleo_l4r5zi/pinmux.c
+++ b/boards/arm/nucleo_l4r5zi/pinmux.c
@@ -63,7 +63,7 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA10, STM32L4X_PINMUX_FUNC_PA10_OTG_FS_ID},
 	{STM32_PIN_PA11, STM32L4X_PINMUX_FUNC_PA11_OTG_FS_DM},
 	{STM32_PIN_PA12, STM32L4X_PINMUX_FUNC_PA12_OTG_FS_DP},
-#endif	/* CONFIG_USB_DC_STM */
+#endif	/* CONFIG_USB_DC_STM32 */
 };
 
 static int pinmux_stm32_init(struct device *port)

--- a/drivers/interrupt_controller/exti_stm32.c
+++ b/drivers/interrupt_controller/exti_stm32.c
@@ -341,7 +341,7 @@ static inline void __stm32_exti_isr_22(void *arg)
 {
 	__stm32_exti_isr(22, 23, arg);
 }
-#endif /* CONFIG_SOC_SERIES_STM32{F4X F7X, F2X} */
+#endif
 #ifdef CONFIG_SOC_SERIES_STM32F7X
 static inline void __stm32_exti_isr_23(void *arg)
 {

--- a/drivers/spi/Kconfig
+++ b/drivers/spi/Kconfig
@@ -160,6 +160,26 @@ config SPI_5_OP_MODES
 
 endif # SPI_5
 
+config SPI_6
+	bool "SPI port 6"
+	help
+	  Enable SPI controller port 6.
+
+if SPI_6
+
+config SPI_6_OP_MODES
+	int "Port 6 supported operation modes (master/slave/both)"
+	default 1
+	range 1 3
+	help
+	  This sets the supported operation modes at runtime, by the SPI
+	  port 6, where:
+	  1 is MASTER mode only (default)
+	  2 is SLAVE mode only
+	  3 is both modes are available.
+
+endif # SPI_6
+
 config SPI_INTEL
 	bool "Intel SPI controller driver"
 	depends on CPU_MINUTEIA

